lib/ansible/plugins/callback/selective.py

```diff
@@ -174,35 +174,45 @@ def v2_playbook_on_task_start(self, task, **kwargs):
 
     def v2_runner_on_ok(self, result, **kwargs):
         """Run when a task finishes correctly."""
-        failed = 'failed' in result._result
-        unreachable = 'unreachable' in result._result
+        failed = result._result.get("failed")
+        unreachable = result._result.get("unreachable")
 
         if 'print_action' in result._task.tags or failed or unreachable or \
-           self._display.verbosity > 1:
+                self._display.verbosity > 1:
             self._print_task()
             self.last_skipped = False
             msg = to_text(result._result.get('msg', '')) or\
-                to_text(result._result.get('reason', ''))
+                to_text(result._result.get('reason', '')) or\
+                to_text(result._result.get('message', ''))
```

Comment:
Remove the portion that adds message as a special key. If there's modules returning message expecting it to be treated specially, change then to return msg instead